src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/TelemetryClientFactory.cs

```diff
@@ -0,0 +1,92 @@
+using Microsoft.ApplicationInsights.WindowsServer;
+using Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    /// <summary>
+    /// Creates a <see cref="TelemetryClient"/> for use by the <see cref="ApplicationInsightsLogger"/>. 
+    /// </summary>
+    [CLSCompliant(false)]
+    public class TelemetryClientFactory : ITelemetryClientFactory
+    {
+        /// <summary>
+        /// Creates a <see cref="TelemetryClient"/>. 
+        /// </summary>
+        /// <returns>The <see cref="TelemetryClient"/> instance.</returns>
+        public TelemetryClient Create(string instrumentationKey, SamplingPercentageEstimatorSettings samplingSettings)
+        {
+            TelemetryConfiguration config = InitializeConfiguration(instrumentationKey, samplingSettings);
+
+            TelemetryClient client = new TelemetryClient(config);
+
+            string assemblyVersion = GetAssemblyFileVersion(typeof(JobHost).Assembly);
+            client.Context.GetInternalContext().SdkVersion = $"webjobs: {assemblyVersion}";
+
+            return client;
+        }
+
+        internal static string GetAssemblyFileVersion(Assembly assembly)
+        {
+            AssemblyFileVersionAttribute fileVersionAttr = assembly.GetCustomAttribute<AssemblyFileVersionAttribute>();
+            return fileVersionAttr?.Version ?? LoggingConstants.Unknown;
+        }
+
+        internal static TelemetryConfiguration InitializeConfiguration(string instrumentationKey,
+            SamplingPercentageEstimatorSettings samplingSettings)
+        {
+            TelemetryConfiguration config = new TelemetryConfiguration();
+            config.InstrumentationKey = instrumentationKey;
+
+            AddInitializers(config);
+
+            // Plug in Live stream and adaptive sampling
+            QuickPulseTelemetryProcessor processor = null;
+            TelemetryProcessorChainBuilder builder = config.TelemetryProcessorChainBuilder
+                .Use((next) =>
+                {
+                    processor = new QuickPulseTelemetryProcessor(next);
+                    return processor;
+                });
+
+            if (samplingSettings != null)
+            {
+                builder.Use((next) =>
+                {
+                    return new AdaptiveSamplingTelemetryProcessor(samplingSettings, null, next);
+                });
+            }
+
+            builder.Build();
+
+            QuickPulseTelemetryModule quickPulse = new QuickPulseTelemetryModule();
+            quickPulse.Initialize(config);
+            quickPulse.RegisterTelemetryProcessor(processor);
+
+            // Plug in perf counters
+            PerformanceCollectorModule perfCounterCollectorModule = new PerformanceCollectorModule();
+            perfCounterCollectorModule.Initialize(config);
+
+            return config;
```

Comment:
since you included ServerTelemetryChannel package - consider overriding `config.TelemetryChannel` with the instance of it for retries and local storage on failure... Do not forget to call `.Initialize` on that channel. Sorry didn't comment on it earlier